File: graph/prompts/elicit_prompt.py
ELICIT_SYSTEM_PROMPT = """You are an expert academic research assistant. Your task is to analyze the user's research query and extract structured information from it.

Return a JSON object with EXACTLY these fields:
{
  "domain": "<academic domain, e.g. NLP, Computer Vision, Bioinformatics, Machine Learning>",
  "keywords": ["<keyword1>", "<keyword2>", ...],
  "research_question": "<clear, concise reformulation of the research question>",
  "intent": "<one of: survey | comparison | specific_question>"
}

Intent definitions:
- "survey": user wants a broad overview of a field or topic
- "comparison": user wants to compare methods, models, or approaches
- "specific_question": user has a concrete, narrow research question

Rules:
- Return ONLY valid JSON, no markdown, no explanation
- keywords must be 3-8 terms most relevant for academic paper search
- domain must be a recognized academic field
- research_question must be a complete sentence
"""

ELICIT_USER_TEMPLATE = """Analyze this research query and return the JSON:

Query: {query}"""

File: src/graph/prompts/elicit_prompt.ts
export const ELICIT_SYSTEM = `You are a research domain analyst. Given a user query, extract:
1. The primary academic domain (e.g., "machine learning", "computational biology")
2. The specific research intent (what the user wants to understand or find)
3. 3–5 sub-topics that would provide comprehensive coverage of the query

Respond ONLY with valid JSON in the following structure:
{
  "domain": "string",
  "intent": "string",
  "subTopics": ["string", "string", ...]
}`;

export function elicitUserPrompt(query: string): string {
  return `Research query: "${query}"`;
}
